cmake/Modules/FindZlib_EP.cmake

```diff
@@ -83,7 +83,9 @@ if (NOT ZLIB_FOUND)
 
     ExternalProject_Add(ep_zlib
       PREFIX "externals"
-      URL "https://downloads.sourceforge.net/project/libpng/zlib/1.2.11/zlib1211.zip"
+      URL
+        "https://github.com/TileDB-Inc/tiledb-deps-mirror/releases/download/2.10-deps/zlib1211.zip"
```

Comment:
I needed a release to attach this to, so that's what I picked for now. My thinking was, whenever we move a dep here (which will be done as-needed), then we'll create a new release in this repo to attach the packages to. We won't change the prior paths ever.